src/term/flavour_py/parser.rs

```diff
@@ -0,0 +1,631 @@
+};
+
+const PREC: &[&[Op]] =
+  &[&[Op::EQL, Op::NEQ], &[Op::LTN], &[Op::GTN], &[Op::ADD, Op::SUB], &[Op::MUL, Op::DIV]];
+
+struct Indent(isize);
+
+impl Indent {
+  fn enter_level(&mut self) {
+    self.0 += 2;
+  }
+
+  fn exit_level(&mut self) {
+    self.0 -= 2;
+  }
+}
+
+pub struct PyParser<'i> {
+  input: &'i str,
+  index: usize,
+}
+
+impl<'a> PyParser<'a> {
+  pub fn new(input: &'a str) -> Self {
+    Self { input, index: 0 }
+  }
+}
+
+impl<'a> ParserCommons<'a> for PyParser<'a> {}
+
+impl<'a> Parser<'a> for PyParser<'a> {
+  fn input(&mut self) -> &'a str {
+    self.input
+  }
+
+  fn index(&mut self) -> &mut usize {
+    &mut self.index
+  }
+}
+
+impl<'a> PyParser<'a> {
+  fn parse_primary_py(&mut self) -> Result<Term, String> {
+    self.skip_trivia();
+    let Some(head) = self.skip_peek_one() else { return self.expected("primary term")? };
+    let res = match head {
+      '(' => {
+        self.consume("(")?;
+        let head = self.parse_term_py()?;
+        if self.skip_starts_with(",") {
+          let mut els = vec![head];
+          while self.try_consume(",") {
+            els.push(self.parse_term_py()?);
+          }
+          self.consume(")")?;
+          Term::Tup { els }
+        } else {
+          self.consume(")")?;
+          head
+        }
+      }
+      '[' => self.list_or_comprehension()?,
+      '\"' => {
+        let str = self.parse_quoted_string()?;
+        let val = STRINGS.get(str);
+        Term::Str { val }
+      }
+      c if c.is_ascii_digit() => {
+        let val = self.parse_u64()?;
+        Term::Num { val: val as u32 }
+      }
+      _ => {
+        if self.try_consume("True") {
+          return Ok(Term::Num { val: 1 });
+        } else if self.try_consume("False") {
+          return Ok(Term::Num { val: 0 });
+        }
+        Term::Var { nam: self.parse_hvml_name()? }
+      }
+    };
+    Ok(res)
+  }
+
+  fn list_or_comprehension(&mut self) -> Result<Term, String> {
+    self.consume("[")?;
+    let head = self.parse_term_py()?;
+    if self.try_consume_keyword("for") {
+      let bind = self.parse_hvml_name()?;
+      self.consume("in")?;
+      let iter = self.parse_term_py()?;
+      let mut cond = None;
+      if self.try_consume_keyword("if") {
+        cond = Some(Box::new(self.parse_term_py()?));
+      }
+      self.consume("]")?;
+      Ok(Term::Comprehension { term: Box::new(head), bind, iter: Box::new(iter), cond })
+    } else {
+      let mut head = vec![head];
+      self.try_consume(",");
+      let tail = self.list_like(|p| p.parse_term_py(), "", "]", ",", true, 0)?;
+      head.extend(tail);
+      Ok(Term::Lst { els: head })
+    }
+  }
+
+  fn parse_term_py(&mut self) -> Result<Term, String> {
+    self.skip_trivia();
+    if self.try_consume_keyword("lambda") {
+      let names = self.list_like(|p| p.parse_hvml_name(), "", ":", ",", true, 1)?;
+      let bod = self.parse_stmt_py(&mut Indent(0))?;
```

Comment:
Confusingly, in python syntax lambdas have an expression as body and not a statement.

Lambdas become a return statement, where the `return` keyword is implicit.